diff --git a/node_modules/ky/distribution/core/Ky.js b/node_modules/ky/distribution/core/Ky.js
index 7d82ba0..a814dfc 100644
--- a/node_modules/ky/distribution/core/Ky.js
+++ b/node_modules/ky/distribution/core/Ky.js
@@ -57,10 +57,12 @@ export class Ky {
                     if (response.status === 204) {
                         return '';
                     }
-                    const arrayBuffer = await response.clone().arrayBuffer();
-                    const responseSize = arrayBuffer.byteLength;
-                    if (responseSize === 0) {
-                        return '';
+                    if(!__DEV__) {
+                        const arrayBuffer = await response.clone().arrayBuffer();
+                        const responseSize = arrayBuffer.byteLength;
+                        if (responseSize === 0) {
+                            return '';
+                        }
                     }
                     if (options.parseJson) {
                         return options.parseJson(await response.text());
diff --git a/node_modules/ky/distribution/utils/delay.js b/node_modules/ky/distribution/utils/delay.js
index 7df9a1f..92aafe1 100644
--- a/node_modules/ky/distribution/utils/delay.js
+++ b/node_modules/ky/distribution/utils/delay.js
@@ -1,7 +1,7 @@
 // https://github.com/sindresorhus/delay/tree/ab98ae8dfcb38e1593286c94d934e70d14a4e111
 export default async function delay(ms, { signal }) {
     return new Promise((resolve, reject) => {
-        if (signal) {
+        if (signal && signal.throwIfAborted) {
             signal.throwIfAborted();
             signal.addEventListener('abort', abortHandler, { once: true });
         }
